Update lib.py

khafrewatkins opened the pull request:
Here are the changes you requested.  Please let me know any corrections or additional alterations need to be made.  I commented out the affected code blocks and adjustments. I know github tracks changes but just to make it more visible.

khafrewatkins committed:
Update lib.py

Here are the changes you requested.  Please let me know any corrections or additional alterations need to be made.  I commented out the affected code blocks and adjustments. I know github tracks changes but just to make it more visible.

siznax commented:
Thanks for the PR @khafrewatkins! :partying_face:

You commented out ``lib.albums()`` (note the "s" on the end—sorry, that's confusing), but we want to leave that function the way it is—so you can _comment_ it back in.

You commented out ``lib.album()`` which is what we want. That function is not being used, and is too close to ``lib.albums()``. So, go ahead and _delete_ that function in your PR code. Don't worry, none of this code will be merged until we're ready.

You changed ``lib.info()`` to ``lib.album()`` which is what I asked for. However, now I see that having ``albums()`` and ``album()`` is confusing. Let's change ``lib.info()``  to ``lib.album_info()`` instead.

If you're still up for it, just make those changes on your PR branch, commit, and push them. They should show up here for us to continue working with.

siznax reviewed (changes requested):
Thanks for the PR @khafrewatkins! :partying_face:

I put my comments in a [PR Review](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-request-reviews)

siznax commented on Moo/play/lib.py:
```diff
@@ -51,46 +51,46 @@
+# def albums(base, index):
+#     ''' Removing/Erasing
+#     returns dict of minimal album metadata
+#     '''
+#     out = dict()
+#     dupes = dict()
+
+#     for ind,path in enumerate(index):
+#         pobj = Path(path)
+#         tracks = len([x for x in pobj.iterdir()])
+
+#         try:
+#             meta = metadata(base, pobj, single=True)['01']
+#         except KeyError:
+#             meta = {'path': path, 'tracks': tracks}
+
+#         year = meta['year'][:4] if meta.get('year') else 'None'
+#         artist = meta.get('album_artist') or meta.get('artist')
+
+#         pruned = {
+#             'album': meta.get('album'),
+#             'artist': artist.split(';')[0] if artist else 'None',
+#             'encoding': meta.get('encoding'),
+#             'genre': meta.get('genre') or 'None',
+#             'index': ind,
+#             'mtime': meta.get('mtime'),
+#             'tracks': tracks,
+#             'year': year}
+
+#         if path in out:
+#             if path not in dupes:
+#                 dupes[path] = list()
+#             dupes[path].append(path)
+#         else:
+#             out[path] = pruned
+
+#     if dupes:
+#         raise ValueError('Found duplicates: {}'.format(dupes))
+
+#     return out
```
You commented out lib.albums() (note the "s" on the end—sorry, that's confusing), but we want to leave that function the way it is—so you can comment it back in.

siznax commented on Moo/play/lib.py:
```diff
@@ -108,13 +108,13 @@ def alpha(albums, mtag='artist'):
-    for path in index:
-        if str(path).endswith(key):
-            return str(path)
+# def album(key, index):
+#     ''' Removing/Erasing
+#     returns album from index matching key
+#     '''
+#     for path in index:
+#         if str(path).endswith(key):
+#             return str(path)
```
You commented out lib.album() which is what we want. That function is not being used, and is too close to lib.albums(). So, go ahead and delete that function. Don't worry, none of this code will be merged until we're ready.

siznax commented on Moo/play/lib.py:
```diff
@@ -354,8 +354,9 @@ def index(config, sort=None):
 
 
-def info(track, metadata):
+def album(track, metadata):
     '''
+    Changed def info to def album()
```
You changed lib.info() to lib.album() which is what I asked for. However, now I see that having albums() and album() is confusing. Let's change lib.info() to lib.album_info() instead.

khafrewatkins commented:
Oh okay got it!  I will make those changes.

On Tue, Mar 2, 2021, 11:33 AM Steve <notifications@github.com> wrote:

> Thanks for the PR @khafrewatkins <https://github.com/khafrewatkins>! 🥳
>
> You commented out lib.albums() (note the "s" on the end—sorry, that's
> confusing), but we want to leave that function the way it is—so you can
> *comment* it back in.
>
> You commented out lib.album() which is what we want. That function is not
> being used, and is too close to lib.albums(). So, go ahead and *delete*
> that function in your PR code. Don't worry, none of this code will be
> merged until we're ready.
>
> You changed lib.info() to lib.album() which is what I asked for. However,
> now I see that having albums() and album() is confusing. Let's change
> lib.info() to lib.album_info() instead.
>
> If you're still up for it, just make those changes on your PR branch,
> commit, and push them. They should show up here for us to continue working
> with.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/siznax/moo/pull/20#issuecomment-789080667>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHF2ONG2CQ4PUY34FGGVA53TBUOM7ANCNFSM4YN72CUA>
> .
>

khafrewatkins committed:
Update lib.py

Here are all the requested changes.  Please let me know if I missed something.  The lib.album is deleted, the lib.albums is no longer commented out and the lib.info is now lib.album_info.

siznax commented on Moo/play/lib.py:
```diff
@@ -51,46 +51,46 @@
+def albums(base, index):
+    ''' Removing/Erasing
+    returns dict of minimal album metadata
+    '''
```
This is  a [docstring](https://www.python.org/dev/peps/pep-0257/) which should not be used for code comments. Let's remove "Removing/Erasing" here.

siznax commented on Moo/play/lib.py:
```diff
@@ -354,9 +347,9 @@ def index(config, sort=None):
 
 
-def album(track, metadata):
+def album_info(track, metadata):
```
Perfect.

siznax commented on Moo/play/lib.py:
```diff
@@ -354,9 +347,9 @@ def index(config, sort=None):
+def album_info(track, metadata):
     '''
-    Changed def info to def album()
+    Changed def info to def album_info()
```
Please remove "Changed def info to def album_info()" from this docstring.

siznax reviewed (changes requested):
Looking good, but we need to keep code change comments out of docstrings. Use the ``-m <msg>`` option for that when you commit. See our project's commit history for examples: https://github.com/siznax/moo/commits/main

khafrewatkins commented:
Ahhh okay.  I will do that know thanks for guiding a newb through this.

On Tue, Mar 2, 2021 at 5:31 PM Steve <notifications@github.com> wrote:

> *@siznax* requested changes on this pull request.
>
> Looking good, but we need to keep code change comments out of docstrings.
> Use the -m <msg> option for that when you commit. See our project's
> commit history for examples: https://github.com/siznax/moo/commits/main
> ------------------------------
>
> In Moo/play/lib.py
> <https://github.com/siznax/moo/pull/20#discussion_r585986715>:
>
> > +    ''' Removing/Erasing
> +    returns dict of minimal album metadata
> +    '''
>
> This is a docstring <https://www.python.org/dev/peps/pep-0257/> which
> should not be used for code comments. Let's remove "Removing/Erasing" here.
> ------------------------------
>
> In Moo/play/lib.py
> <https://github.com/siznax/moo/pull/20#discussion_r585987028>:
>
> > @@ -354,9 +347,9 @@ def index(config, sort=None):
>      return sorted(albums, key=sort_key, reverse=True)
>
>
> -def album(track, metadata):
> +def album_info(track, metadata):
>
> Perfect.
> ------------------------------
>
> In Moo/play/lib.py
> <https://github.com/siznax/moo/pull/20#discussion_r585987998>:
>
> >      '''
> -    Changed def info to def album()
> +    Changed def info to def album_info()
>
> Please remove this from the docstring. Git will show us this change.
> Comments like this should be added as a -m <msg>
> <https://git-scm.com/docs/git-commit#Documentation/git-commit.txt--mltmsggt>
> to your commit.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/siznax/moo/pull/20#pullrequestreview-602367472>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHF2ONAZTDZBCPEHK2L42C3TBVYLNANCNFSM4YN72CUA>
> .
>

khafrewatkins committed:
Update lib.py

The comments I added are removed now.

siznax commented:
Thanks for your help @khafrewatkins. It looks like we lost the entire docstring from ``lib.albums()``. We'll have to restore that. I can't merge this pull request at this point, because it will break the main branch. Once I start developing unit tests, that will be apparent in GitHub.

Remember, there is no such thing as a "newbie" because no one can know everything. We all have something to learn _from each other_.

Let me know if you want to continue working on this. We need to satisfy all of #15 in order to merge.

We're _almost_ there 😃

khafrewatkins commented:
Yes I will do whatever needs to be done!

On Wed, Mar 3, 2021, 10:29 AM Steve <notifications@github.com> wrote:

> Thanks for your help @khafrewatkins <https://github.com/khafrewatkins>.
> It looks like we lost the entire docstring from lib.albums(). We'll have
> to restore that. I can't merge this pull request at this point, because it
> will break the main branch. Once I start developing unit tests, that will
> be apparent in GitHub.
>
> Remember, there is no such thing as a "newbie" because no one can know
> everything. We all have something to learn *from each other*.
>
> Let me know if you want to continue working on this. We need to satisfy
> all of [#15 <https://github.com/siznax/moo/issues/15>] in order to merge.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/siznax/moo/pull/20#issuecomment-789847072>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHF2ONDJUUS4YWZXH53TGGLTBZPVTANCNFSM4YN72CUA>
> .
>

siznax commented on Moo/play/lib.py:
```diff
@@ -52,7 +52,7 @@
 
 def albums(base, index):
-    '''
+    ''' Removing/Erasing
```
Please remove "Removing/Erasing" but leave the docstring.

khafrewatkins committed:
Update lib.py

The docstring has been returned.  
'''
    returns dict of minimal album metadata
'''

siznax commented:
Thanks for your help @khafrewatkins!
